src/components/MermaidDash.js

```diff
@@ -244,6 +240,27 @@ class MermaidDash extends Component {
     this.resize();
   }
 
+  fetchSitesChunk = async (params, pageNo = 1) => {
+    let siteResults = {};
+    const apiResults = await summary.get('/sites/', {
+      params: {
+        page: pageNo,
+        ...params
+      }
+    });
+    siteResults = apiResults;
+    return Object.keys(siteResults).length === 0 ? [] : siteResults.data.features;
+  };
+
+  fetchEntiresSites = async (params, pageNo = 1) => {
+    const results = await this.fetchSitesChunk(params, pageNo);
+    if (results.length === this.state.queryLimit) {
+      return [...results].concat(await this.fetchEntiresSites(params, pageNo + 1));
+    } else {
+      return results;
+    }
+  };
```

Comment:
I'm curious about how this particular bit of recursion is implemented. I'm looking at line `257` where you're doing another call only if `results.length === this.state.queryLimit`.

Can you explain what happens if the number of results in the last page coincides with the limit? Will another request happen, and if so, what happens if you fetch another query with no results? 

Ideally, the best practice is to minimize api calls. Perhaps there's an opportunity to jam in a "total records" parameter in the API response that'll help out here...